ecs-agent/netlib/platform/common_linux.go

```diff
@@ -609,6 +623,71 @@ func (c *common) configureBranchENI(ctx context.Context, netNSPath string, eni *
 	return err
 }
 
+func (c *common) configureGENEVEInterface(
+	ctx context.Context,
+	netNSPath string,
+	iface *networkinterface.NetworkInterface,
+	netDAO netlibdata.NetworkDataClient,
+) error {
+	var cniNetConf ecscni.PluginConfig
+	add := true
+
+	// Generate CNI network configuration based on the ENI's desired state.
+	switch iface.DesiredStatus {
+	case status.NetworkReadyPull:
+		// Assign destination port and set device name for V2N ENI.
+		if err := ecscni.SetV2NDstPortAndDeviceName(iface, netDAO); err != nil {
+			return err
+		}
+		cniNetConf = NewTunnelConfig(netNSPath, iface, VPCTunnelInterfaceTypeGeneve)
+	case status.NetworkReady:
+		cniNetConf = NewTunnelConfig(netNSPath, iface, VPCTunnelInterfaceTypeTap)
+	case status.NetworkDeleted:
```

Comment:
Seems like these two cases are doing the same thing. Instead of 
```
case a:
case b:
```
can we merge them together by doing `case a, b: `?

Reply:
The 3rd argument is different for both cases.